src/agents/utility.jl

```diff
@@ -2218,6 +2218,55 @@ function solve_agent_problem!(
         for y in model_data.index_y, l in utility.index_l
             utility.flow_cap_my(y, l, :) .= value.(flow_cap[y, l])
         end
+
+        for y in model_data.index_y,
+            s in utility.index_stor_existing,
+            z in model_data.index_z,
+            d in model_data.index_d,
+            t in model_data.index_t
+    
+            utility.charge_E_my(y, s, z, d, t, :) .= value.(charge_E[y, s, z, d, t])
+        end
+        for y in model_data.index_y,
+            s in utility.index_stor_existing,
+            z in model_data.index_z,
+            d in model_data.index_d,
+            t in model_data.index_t
+    
+            utility.discharge_E_my(y, s, z, d, t, :) .= value.(discharge_E[y, s, z, d, t])
+        end
+        for y in model_data.index_y,
+            s in utility.index_stor_new,
+            z in model_data.index_z,
+            d in model_data.index_d,
+            t in model_data.index_t
+    
+            utility.charge_C_my(y, s, z, d, t, :) .= value.(charge_C[y, s, z, d, t])
+        end
+        for y in model_data.index_y,
+            s in utility.index_stor_new,
+            z in model_data.index_z,
+            d in model_data.index_d,
+            t in model_data.index_t
+    
+            utility.discharge_C_my(y, s, z, d, t, :) .= value.(discharge_C[y, s, z, d, t])
+        end
+        for y in model_data.index_y,
+            s in utility.index_stor_existing,
+            z in model_data.index_z,
+            d in model_data.index_d,
+            t in model_data.index_t
+    
+            utility.energy_E_my(y, s, z, d, t, :) .= value.(energy_E[y, s, z, d, t])
+        end
+        for y in model_data.index_y,
+            s in utility.index_stor_new,
+            z in model_data.index_z,
+            d in model_data.index_d,
+            t in model_data.index_t
+    
+            utility.energy_C_my(y, s, z, d, t, :) .= value.(energy_C[y, s, z, d, t])
+        end
```

Comment:
```suggestion
        for y in model_data.index_y,
            s in utility.index_stor_existing,
            z in model_data.index_z,
            d in model_data.index_d,
            t in model_data.index_t
    
            utility.charge_E_my(y, s, z, d, t, :) .= value.(charge_E[y, s, z, d, t])
            utility.discharge_E_my(y, s, z, d, t, :) .= value.(discharge_E[y, s, z, d, t])
            utility.charge_C_my(y, s, z, d, t, :) .= value.(charge_C[y, s, z, d, t])
            utility.discharge_C_my(y, s, z, d, t, :) .= value.(discharge_C[y, s, z, d, t])
            utility.energy_E_my(y, s, z, d, t, :) .= value.(energy_E[y, s, z, d, t])
            utility.energy_C_my(y, s, z, d, t, :) .= value.(energy_C[y, s, z, d, t])
        end
```

Reply:
Why wouldn't you write it like this? Are there other outputs with the same indices whose reports should also be collapsed?

Reply:
I updated the code to collapse some index (`s` actually have different index: `utility.index_stor_existing` and `utility.index_stor_new` for `E` and `C` so they have to be separate).